pkg/services/baremetal/client/ssh/ssh_client.go

```diff
@@ -415,32 +458,34 @@ EOF_VIA_SSH`, cmd))
 		return out
 	}
 
-	installImageOut := c.runSSH(`sh /root/installimage-wrapper.sh`)
+	return c.runSSH(`nohup /root/installimage-wrapper.sh >/root/installimage-wrapper.sh.log 2>&1 </dev/null &`)
```

Comment:
I have recently discovered `systemd-run` for this usecase. You get some nice goodies for free:

- logs in journald
- Process is automatically executed in the background (no need for `nohup &`)
- You can easily check the status through `systemctl status task-name.service`

Reply:
@apricote thank you for the hint `systemd-run` is new to me. The above code runs in the rescue-system. Is systemd available in the rescue-system? Overall all I am happy with `nohup`. But thank you for the hint!

Reply:
At least on the Cloud rescue system `systemd` is used. As far as I understood Robot Rescue uses the same OS.

Currently installed version is `systemd 252` and `systemd-run` is available.

Reply:
I tested `systemd-run` now.

```
root@rescue ~ # cat t.sh 
#!/bin/bash
echo "foo stdout"
echo "foo stderr" >&2
```

```
root@rescue ~ # systemd-run /root/t.sh 
Running as unit: run-r7e29081996164cb88dd2dff8bba7c2d3.service
```

```
root@rescue ~ # journalctl -u run-r7e29081996164cb88dd2dff8bba7c2d3.service
Aug 07 08:41:33 rescue systemd[1]: Started run-r7e29081996164cb88dd2dff8bba7c2d3.service - /root/t.sh.
Aug 07 08:41:33 rescue t.sh[52388]: foo stdout
Aug 07 08:41:33 rescue t.sh[52388]: foo stderr
Aug 07 08:41:33 rescue systemd[1]: run-r7e29081996164cb88dd2dff8bba7c2d3.service: Deactivated successfully.
```

I would prefer `nohup` at the moment, because it is easier to get to the logs.

With `systemd-run` I need to parse the temporary service-name (`run-r7e290819...`).

@apricote or do you see a big benefit in using `systemd-run`, which I don't see at the moment?

Reply:
@apricote I resolve this conversation, otherwise I can't merge it. We can switch to systemd-run, if there are good reasons in a second PR.

Reply:
You can set the name when calling 'systemd-run' with a flag. That way you don't have to parse and keep the random name around.

I don't have a strong opinion here, happy if it works with either solution. I don't use the bare metal part of CAPH.